python/lib/dependabot/python/update_checker/poetry_version_resolver.rb

```diff
@@ -298,12 +298,32 @@ def update_dependency_requirement(toml_node, requirement)
         end
 
         def subdep_type
```

Comment:
I just happened to find this function unexpectedly searching around for `subdep_type`. Looks like there is a decent amount of logic shared in the file updater, but probably small enough such that it made more sense to copy into here vs extract into a utility.

Since this is getting larger, maybe it makes sense to refactor to a utility? Another set of options:

1. Refactor into a utility function and call from both places. Opinion on where this should go?
2. Remove this modification since it's not specifically tackling the core issue and we are trying to be pretty targeted per other conversations
3. Keep as-is

Reply:
You forgot to pass `poetry_object` around, and a real update seems to be still failing because of that. So this needs to be fixed before merging, and also signals some lack of coverage here.

I'll look into a good place for factoring out this method.

Reply:
I proposed https://github.com/dependabot/dependabot-core/pull/7418 to refactor this. We can rebase and adapt this PR to it, and should be all good.